### PR TITLE
feat: add node engines to upgrade script

### PR DESCRIPTION
**Pull request by mkilpatrick**
Adds supported engines to package.json

```
"engines": {
    "node": "^18.0.0 || >=20.0.0"
  },
```

### CODE REVIEW

File: packages/pages/src/upgrade/pagesUpdater.ts [hunk 1]
```diff
@@ -118,6 +118,25 @@ const installDependencies = async (
...
+const updatePackageEngines = (targetDirectory: string) => {
+  const packageJsonPath = path.resolve(targetDirectory, "package.json");
+
+  try {
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, "utf-8"));
+    let engines = packageJson.engines;
+    if (!engines) {
+      engines = { node: "" };
+    }
+    engines.node = "^18.0.0 || >=20.0.0";
```
Comment (jwartofsky-yext) [inline packages/pages/src/upgrade/pagesUpdater.ts:131]: Kind of strange that we support 18 and 20+, but not 19

Comment (mkilpatrick) [reply packages/pages/src/upgrade/pagesUpdater.ts:131]: We are support what Vite supports. Odd versions of Node are quick betas. Once the next even version is released the prior odd becomes deprecated, with the previous even being in LTS.

Comment (mkilpatrick) [reply packages/pages/src/upgrade/pagesUpdater.ts:131]: https://nodejs.org/en/about/previous-releases

File: packages/pages/src/upgrade/pagesUpdater.ts [hunk 2]
```diff
@@ -118,6 +118,25 @@ const installDependencies = async (
...
 };
 
+// Function to update package.json engines
+const updatePackageEngines = (targetDirectory: string) => {
+  const packageJsonPath = path.resolve(targetDirectory, "package.json");
+
+  try {
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, "utf-8"));
+    let engines = packageJson.engines;
+    if (!engines) {
```
Comment (jwartofsky-yext) [inline packages/pages/src/upgrade/pagesUpdater.ts:128]: Could also use:

NODE_ENGINE = "^18.0.0 || >=20.0.0";
if (engines) {
  engines.node = NODE_ENGINE;
else {
  engines = { node: NODE_ENGINE}
}

**Review by jwartofsky-yext — APPROVED** [review]

**Review by jwartofsky-yext — APPROVED** [review]
lgtm

**Pull request merged**